plaso/parsers/shared/unified_logging/tracev3.py

```diff
@@ -0,0 +1,1221 @@
+# -*- coding: utf-8 -*-
+"""The Apple Unified Logging (AUL) TraceV3 file parser."""
+
+import binascii
+import decimal
+import ipaddress
+import os
+import re
+import stat
+
+import lz4.block
+
+from dfdatetime import posix_time as dfdatetime_posix_time
+
+from plaso.helpers.macos import darwin
+from plaso.helpers.macos import dns
+from plaso.helpers.macos import location
+from plaso.helpers.macos import opendirectory
+from plaso.helpers.macos import tcp
+from plaso.helpers import sqlite
+
+from plaso.lib.aul import constants
+from plaso.lib.aul import time as aul_time
+
+from plaso.lib import dtfabric_helper
+from plaso.lib import errors
+
+from plaso.parsers.shared.unified_logging import activity
+from plaso.parsers.shared.unified_logging import dsc
+from plaso.parsers.shared.unified_logging import loss
+from plaso.parsers.shared.unified_logging import nonactivity
+from plaso.parsers.shared.unified_logging import oversize
+from plaso.parsers.shared.unified_logging import signpost
+from plaso.parsers.shared.unified_logging import simpledump
+from plaso.parsers.shared.unified_logging import statedump
+from plaso.parsers.shared.unified_logging import trace
+
+from plaso.parsers import interface
+from plaso.parsers import logger
+
+
+class TraceV3FileParser(interface.FileObjectParser,
+                        dtfabric_helper.DtFabricHelper):
+  """Apple Unified Logging and Activity Tracing (tracev3) file."""
+
+  _DEFINITION_FILE = os.path.join(
+      os.path.dirname(__file__), 'aul.yaml')
+
+  _CATALOG_LZ4_COMPRESSION = 0x100
+
+  # Chunk Tags
+  _CHUNK_TAG_HEADER = 0x1000
+  _CHUNK_TAG_FIREHOSE = 0x6001
+  _CHUNK_TAG_OVERSIZE = 0x6002
+  _CHUNK_TAG_STATEDUMP = 0x6003
+  _CHUNK_TAG_SIMPLEDUMP = 0x6004
+  _CHUNK_TAG_CATALOG = 0x600B
+  _CHUNK_TAG_CHUNKSET = 0x600D
+
+  format_strings_re = re.compile(
+      r'%(\{[^\}]{1,64}\})?([0-9.'
+      r' *\-+#\']{0,6})([hljztLq]{0,2})([@dDiuUxXoOfeEgGcCsSpaAFPm])'
+  )
+
+  def __init__(self, timesync_parser, uuid_parser, dsc_parser):
+    """Initializes a tracev3 file parser."""
+    super(TraceV3FileParser, self).__init__()
+    self.boot_uuid_ts = None
+    self.catalog = None
+    self.catalog_files = []
+    self.chunksets = []
+    self.dsc_parser = dsc_parser
+    self.header = None
+    self.logs = []
+    self.oversize_data = []
+    self.timesync_parser = timesync_parser
+    self.uuid_parser = uuid_parser
+
+  def GetDataTypeMap(self, name):
+    """Wrapper around dtfabric's _GetDataTypeMap()"""
+    return self._GetDataTypeMap(name)
+
+  def ReadStructureFromByteStream(self, data, offset, data_map):
```

Comment:
FYI if you have to make wrappers like this, it is often a good indication that concerns are not cleanly separated